### PR TITLE
Refine State trie migration 

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR improves state trie migration like below.
- fix a minor bug that `admin_stateMigrationStatus()` shows not 100% progress after state migration.
- expand the lock scope to prevent a conflict of reading/change state db.
- apply concurrent removing database to reduce the delay for big database

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage/database/db_manager.go [hunk 1]
```diff
@@ -648,13 +648,17 @@ func (dbm *databaseManager) CreateMigrationDBAndSetStatus(blockNum uint64) error
 
 	// Store the migration status
 	dbm.setStateTrieMigrationStatus(blockNum)
+	dbm.lockInMigration.Unlock()
```
Comment (ehnuje) [inline storage/database/db_manager.go:651]: It would be better to call `Unlock` with `defer` to avoid unexpected situation in the future.

File: storage/database/db_manager.go [hunk 2]
```diff
@@ -593,8 +593,6 @@ func (dbm *databaseManager) setStateTrieMigrationStatus(blockNum uint64) {
 	if err := miscDB.Put(migrationStatusKey, common.Int64ToByteBigEndian(blockNum)); err != nil {
 		logger.Crit("Failed to set state trie migration status", "err", err)
 	}
-	dbm.lockInMigration.Lock()
-	defer dbm.lockInMigration.Unlock()
```
Comment (ehnuje) [inline storage/database/db_manager.go:597]: Why are these lines removed?

Comment (KimKyungup) [reply storage/database/db_manager.go:597]: To expand the scope of the lock, I moved this to callee function.
See line 641. :)

**Review by ehnuje — APPROVED** [review]

File: storage/database/db_manager.go [hunk 3]
```diff
@@ -640,6 +638,9 @@ func (dbm *databaseManager) CreateMigrationDBAndSetStatus(blockNum uint64) error
 	// Create a new database for migration process.
 	newDB, newDBDir := newStateTrieMigrationDB(dbm.config, blockNum)
 
+	dbm.lockInMigration.Lock()
+	defer dbm.lockInMigration.Unlock()
```
Comment (yoomee1313) [inline storage/database/db_manager.go:642]:
```suggestion
	// lock to prevent from a conflict of reading state DB and changing state DB
	dbm.lockInMigration.Lock()
	defer dbm.lockInMigration.Unlock()
```
To be sure, how about adding a comment like the code below?

**Review by winnie-byun — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]
